bokehjs/src/lib/models/glyphs/line.ts

```diff
@@ -43,17 +44,30 @@ export class LineView extends XYGlyphView {
       const sx_i = sx[i]
       const sy_i = sy[i]
 
+      if (nonselection && !move && iprev != null && i-iprev > 1 && isFinite(sx[iprev+1] + sy[iprev+1]))
+        ctx.lineTo(sx[iprev+1], sy[iprev+1])  // End of previous line
+
       if (!isFinite(sx_i + sy_i))
         move = true
       else {
         if (move || gap(i)) {
-          ctx.moveTo(sx_i, sy_i)
+          if (nonselection && i > 0 && isFinite(sx[i-1] + sy[i-1])) {
+            ctx.moveTo(sx[i-1], sy[i-1])  // Start of new line
+            ctx.lineTo(sx_i, sy_i)
+          } else
+            ctx.moveTo(sx_i, sy_i)
           move = false
         } else
           ctx.lineTo(sx_i, sy_i)
+
+        iprev = i
       }
+    }
 
-      last_i = i
+    if (nonselection && !move && iprev != null) {
+      const n = sx.length
```

Comment:
not really

Reply:
At this stage arrays (uniforms) are supposed to be of equal length, so doing what you did is fine. However, `GlyphView.data_size` exists and will give you the computed size of the data. On the other hand, do not use `ColumnDataSource.get_length()`, because of scalar values and generated data and what not.

Reply:
OK, thanks.